src/index.js

```diff
@@ -305,113 +279,103 @@ var prepare = {
   /**
    */
 
-  $ne: function(a) {
+  $ne(a) {
     return prepare.$eq(a);
   },
 
   /**
    */
 
-  $and: function(a, query, options) {
-    return a.map(parse(options));
+  $and(a, query, parseSub) {
+    return a.map(parseSub);
   },
 
   /**
    */
 
-  $all: function(a) {
-    return prepare.$and(a);
+  $all(a, query, parseSub) {
+    return prepare.$and(a, query, parseSub);
```

Comment:
Oh interesting, where did these extra parameters come from?

Reply:
They're used in `$and` and always passed in from the `parse` function - this was apparently a preexisting bug because it started failing tests when I tweaked a few things.

Reply:
oooh